### DISCUSSION TITLE
Feature Request: In-game switch for turning off hitsounds

### DISCUSSION THREAD

**Discussion by kisaragi-hiu** [ppy/osu#13002] [Ideas]
**Describe the feature:**

Currently there is no way to disable hitsounds without also turning off UI effects. 

Sometimes I don't want gameplay feedback because I want to hear the music more, but the only option right now is using a skin with silent hitsounds. I propose a new switch in the pre-game screen to turn off hitsounds.

**Proposal designs of the feature:**
![untitled](https://user-images.githubusercontent.com/11722318/48302595-2e733980-e539-11e8-8bf5-052cef189a56.png)

**Comment by UselessToucan** [comment]
Why do you need UI effects during gameplay? Do they even exist?

**Comment by Aergwyn** [comment]
It's about splitting hitsounds and everything else that falls under that category. Another example is disabling menu sound without removing hitsounds.

**Comment by kisaragi-hiu** [comment]
@Aergwyn I'd imagine that hitsounds still depends on UI effects volume, Turning off UI effects (by setting volume to 0) without removing hitsounds is not what I'm requesting.

@UselessToucan Not during gameplay, but for example the pause menu would stay silent until I manually turn up effects volume again.

**Comment by Aergwyn** [comment]
@kisaragi-hiu What I said is the same as yours just inversed. Both would depend on splitting hit sound and menu sound volumes.  
While your suggestion is just a toggle I'd propose having it like any other volume setting.

**Comment by UselessToucan** [comment]
Which is the prefered way to resolve this volume separation?
1. Add one more 'SampleManager' to the `AudioManager`? (`GameplaySampleManager`)
2. Set both volumes to samples using single `SampleManager`? (Probably using Strategy pattern under the hood)

**Comment by LastExceed** [comment]
@UselessToucan there's also 1 more thing during gameplay that's controlled by the sound effects volume, and that's the nightcore beats (they're not implemented yet). I could imagine people wanting to mute those while keeping the hitsounds, so it makes sense to separate them. Sure, that could be done with skinning (using silent sound files), but that would also apply to the OP.

**Comment by peppy** [comment]
Strongly against this. I’d argue that if you really want to destroy the game like this, using a skin is the correct method. I’d rather not add an option like this in the user’s face when it takes away what makes osu! osu!.

**Comment by Aergwyn** [comment]
>Strongly against this. 

 In general or regarding to muting nightcore beats?

**Comment by smoogipoo** [comment]
Would be ok with a volume meter for hitsounds, probably even replacing the effects meter during gameplay or something.... But marking this as low priority for now since the effects meter is usable.
